src/common/ZulipTextButton.js

```diff
@@ -98,21 +105,37 @@ type Props = $ReadOnly<{|
 // things like project-specific styles and making any sensible adjustments
 // to the interface.
 export default function ZulipTextButton(props: Props): Node {
-  const { variant = 'standard', leftMargin, rightMargin, label, onPress } = props;
+  const { variant = 'standard', leftMargin, rightMargin, label, onPress, disabled } = props;
 
   const variantStyles = useMemo(() => styleSheetForVariant(variant), [variant]);
 
+  const isNotDisabled = disabled !== true;
+
+  const disabledGate = useCallback(
+    onPressCallback => () => {
+      if (isNotDisabled) {
+        onPressCallback();
+      }
+    },
+    [isNotDisabled],
+  );
+
+  const buttonDisabledColor = useMemo(
```

Comment:
I'm wondering if there should be a ternary conditional inside `useMemo`. In retrospect, the call should just memoize a value provided, without it having to do further operations.